code/modules/antagonists/vampire/vampire_powers/vampire_powers.dm

```diff
@@ -431,34 +433,7 @@
 
 /obj/effect/proc_holder/spell/targeted/enthrall/proc/handle_enthrall(mob/living/user, mob/living/carbon/human/H)
 	if(!istype(H))
-		return 0
-	var/ref = "\ref[user.mind]"
-	if(!(ref in SSticker.mode.vampire_thralls))
-		SSticker.mode.vampire_thralls[ref] = list(H.mind)
-	else
-		SSticker.mode.vampire_thralls[ref] += H.mind
-	SSticker.mode.update_vampire_icons_added(H.mind)
-	SSticker.mode.update_vampire_icons_added(user.mind)
-	var/datum/mindslaves/slaved = user.mind.som
-	if(!slaved)
-		slaved = new()
-		slaved.masters = user.mind
-	H.mind.som = slaved
-	slaved.serv += H
-	slaved.add_serv_hud(user.mind, "vampire")//handles master servent icons
-	slaved.add_serv_hud(H.mind, "vampthrall")
-
-	SSticker.mode.vampire_enthralled.Add(H.mind)
-	SSticker.mode.vampire_enthralled[H.mind] = user.mind
-	H.mind.special_role = SPECIAL_ROLE_VAMPIRE_THRALL
-
-	var/datum/objective/protect/serve_objective = new
-	serve_objective.owner = user.mind
-	serve_objective.target = H.mind
-	serve_objective.explanation_text = "You have been Enthralled by [user.real_name]. Follow [user.p_their()] every command."
-	H.mind.objectives += serve_objective
-
-	to_chat(H, "<span class='biggerdanger'>You have been Enthralled by [user.real_name]. Follow [user.p_their()] every command.</span>")
-	to_chat(user, "<span class='warning'>You have successfully Enthralled [H]. <i>If [H.p_they()] refuse[H.p_s()] to do as you say just adminhelp.</i></span>")
```

Comment:
These lines should probably be kept for the feedback. And to avoid avoid double `to_chat`s for the thrall, you can put the "You have been Enthralled by..." as the thrall datum's `_greet_text` arg for it's `/New`. So for example:
```dm
var/greet_text = "You have been Enthralled by [user.real_name]. Follow [user.p_their()] every command."
H.mind.add_antag_datum(new /datum/antagonist/mindslave/thrall(user.mind, greet_text))
```